Install dependencies as project dependencies from NPM repo

jstirnaman opened the pull request:
…endencies. Run npm install or yarn install. (#2474)

Closes #2474 

_Describe your proposed changes here._

- [x] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/docs-v2/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

sanderson reviewed (changes requested):
`package.json` looks good. There are a few other places we need to update:

- Content in the `README.md` about running the docs locally.
- The deploy script in `circleci/config.yaml` (specifically lines 24-26)

kelseiv commented on README.md:
````diff
@@ -20,37 +23,31 @@ including our GPG key, can be found at https://www.influxdata.com/how-to-report-
+          ```
+          $ cd docs-v2
+          $ yarn install
+          ```
````
not sure if it makes sense to tell folks to delete package.json, but I get the following error (which may be trivial?) when testing:
```kseivert@MBP15-KSEIVERT docs-v2 % yarn install
yarn install v1.22.10
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
```

sanderson replied:
This is trivial. This just means that you likely used npm to generate assets in your local repo. For fresh clones of the docs repo, this won't matter.

sanderson commented on README.md:
````diff
@@ -20,37 +23,31 @@ including our GPG key, can be found at https://www.influxdata.com/how-to-report-
 
+      ```
+      $ cd docs-v2/
+      $ hugo server
````
With hugo loaded as a node dependency, we should probably change this to use npx:


```suggestion
      $ npx hugo server
```

sanderson commented on package.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "devDependencies": {
+    "autoprefixer": "^10.2.5",
+    "hugo-extended": "^0.83.1",
+    "postcss": "^8.2.13",
+    "postcss-cli": "^8.3.1"
+  }
+}
```
Since we no longer load dependencies globally in the deploy script, we need to also include `redoc-cli` as a dependency. Version `0.9.13` is the only version I've found to consistently generate the API docs.

```suggestion
{
  "devDependencies": {
    "autoprefixer": "^10.2.5",
    "hugo-extended": "^0.83.1",
    "postcss": "^8.2.13",
    "postcss-cli": "^8.3.1",
    "redoc-cli": "0.9.13"
  }
}
```

sanderson reviewed (changes requested):
A few suggestions in the existing changes. We also need to update the API generation script. I'll do that and push a commit.

sanderson commented on README.md:
````diff
@@ -20,37 +23,31 @@ including our GPG key, can be found at https://www.influxdata.com/how-to-report-
 
-    View the docs at [localhost:1313](http://localhost:1313).
+      ```
+      $ cd docs-v2/
````
Since they should already be in the docs-v2 directory, we can drop this command.

sanderson reviewed (approved):
One minor suggestion and then I think it's good. We'll need to be sure that the API docs generate correctly after we merge this change. Sometimes the generation breaks, but it doesn't throw an error.

jstirnaman commented on api-docs/generate-api-docs.sh:
```diff
@@ -41,7 +41,7 @@ weight: 304
 
   # Use Redoc to generate the v2 API html
-  redoc-cli bundle -t template.hbs \
+  npx redoc-cli bundle -t template.hbs \
```
@sanderson Should we give `/api-docs` it's own package.json as well? I also need to update the README with npx.

sanderson replied:
That's not a bad idea. That way those who don't need to generate the API docs won't need to download all the deps to do it. We'll need to update the deploy steps `circleci/config.yml` as well.

jstirnaman replied:
Done

jstirnaman commented:
> One minor suggestion and then I think it's good. We'll need to be sure that the API docs generate correctly after we merge this change. Sometimes the generation breaks, but it doesn't throw an error.

Please review again. Added the package.json to api-docs and updated the ci script accordingly.

sanderson reviewed (approved):
LGTM. Thanks for this @jstirnaman!
